robosherlock/src/flowcontrol/src/RSProcessManager.cpp

```diff
@@ -287,6 +291,113 @@ bool RSProcessManager::handleQuery(std::string &request, std::vector<std::string
+          result.push_back("Tracking stopped.");
+          return true;
+          }
+        }
+      }
+      if(!waitForServiceCall_)
+      {
+        result.push_back("There's another tracking task being handled right now. Please stop it before starting tracking anew.");
+        return true;
+      }
+      outInfo(FG_GREEN << "START TRACKING");
+      outInfo(FG_BLUE << "Executing Pipeline # generated by query");
+      visualizer_.setActiveAnnotators(new_pipeline_orders[0]);
+
+      outInfo("Resetting CAS...");
+      engine_->resetCas();
+      rs::Query query = rs::create<rs::Query>(*engine_->getCas());
+      query.query.set(request);
+      rs::SceneCas sceneCas(*engine_->getCas());
+      sceneCas.set("QUERY", query);
+
+      engine_->setPipelineOrdering(new_pipeline_orders[0]);
+      engine_->processOnce();
+      //outInfo("Resetting CAS...");
+      //engine_->resetCas();
+      engine_->resetPipelineOrdering();
+      engine_->setPipelineOrdering(lowLvlPipeline_);
+
+      std::vector<std::string> result_designators;
+      rs::ObjectDesignatorFactory dw(engine_->getCas());
+      use_identity_resolution_ ? dw.setMode(rs::ObjectDesignatorFactory::Mode::OBJECT) : dw.setMode(rs::ObjectDesignatorFactory::Mode::CLUSTER);
+      dw.getObjectDesignators(result_designators);
+
+      for(int a = 0; a < result_designators.size(); a++){
+        outInfo(result_designators[a]);
+      }
+      outInfo("result_designators has ended.");
+
+      std::vector<std::string> filtered_response;
+      std::vector<bool> designators_to_keep;
+      query_interface->filterResults(result_designators, filtered_response, designators_to_keep);
+      int obj_id = -1;
+      for(int n = 0; n < designators_to_keep.size(); n++)
+      {
+        if(designators_to_keep[n])
+        {
+          obj_id = n;
+          outInfo("Object " + std::to_string(obj_id) +  " satisfies query designators.");
+          break;
+        }
+      }
+
+      if(obj_id == -1)
+      {
+        outError("Can't find any object fulfilling all given constraints. Tracking will not be started.");
+        engine_->reconfigure();
+        this->waitForServiceCall_=true;
+        result.push_back("Can't find any object fulfilling all given constraints. Tracking has not been started.");
+        return true;
+      }
+      else
+        {
+        uima::CAS *tcas = engine_->getCas();
+        rs::SceneCas sceneCas(*tcas);
+        rs::Scene scene = sceneCas.getScene();
+
+        std::vector<rs::ObjectHypothesis> obj_hyps;
+        scene.identifiables.filter(obj_hyps);
+        rs::Size s = rs::create<rs::Size>(*tcas); // This is just a hack to get a simple integer (the object ID) into the cas.
+        s.height.set(obj_id);
+        outInfo(FG_GREEN << "SETTING OBJ_TO_TRACK");
+        sceneCas.setFS("OBJ_ID_TRACK", s);
+        outInfo("Found " << obj_hyps.size() << " objects");
+
+        visualizer_.setActiveAnnotators(new_pipeline_orders[1]);
```

Comment:
what if new_pipeline_order is only of size 1? check size of vect before accessing;